File: Models/Dispute.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ShopNest.Models
{
    public class Dispute
    {
        [Key]
        public int DisputeId { get; set; }

        [Required]
        public int OrderId { get; set; }

        [Required]
        public int CustomerId { get; set; }

        [Required]
        public int SellerId { get; set; }

        [Required]
        [StringLength(200)]
        public string Subject { get; set; } = string.Empty;

        [Required]
        [StringLength(2000)]
        public string Description { get; set; } = string.Empty;

        [Required]
        public DisputeStatus Status { get; set; } = DisputeStatus.Open;

        [Required]
        public DisputeType Type { get; set; }

        public int? AssignedAdminId { get; set; }

        [StringLength(2000)]
        public string? Resolution { get; set; }

        public DateTime? ResolvedAt { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [ForeignKey("OrderId")]
        public virtual Order Order { get; set; } = null!;

        [ForeignKey("CustomerId")]
        public virtual User Customer { get; set; } = null!;

        [ForeignKey("SellerId")]
        public virtual User Seller { get; set; } = null!;

        [ForeignKey("AssignedAdminId")]
        public virtual User? AssignedAdmin { get; set; }

        public virtual ICollection<DisputeMessage> DisputeMessages { get; set; } = new List<DisputeMessage>();
    }

    public enum DisputeStatus
    {
        Open = 1,
        InProgress = 2,
        Resolved = 3,
        Closed = 4,
        Escalated = 5
    }

    public enum DisputeType
    {
        ItemNotReceived = 1,
        ItemNotAsDescribed = 2,
        DamagedItem = 3,
        RefundIssue = 4,
        ShippingIssue = 5,
        Other = 6
    }
}

File: app/Models/Dispute.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use App\Enums\DisputesStatus;
use App\Enums\DisputeType;

class Dispute extends Model
{
    protected $primaryKey = 'dispute_id';

    protected $fillable = [
        'order_id',
        'customer_id',
        'seller_id',
        'subject',
        'description',
        'status',
        'type',
        'assigned_admin_id',
        'resolution',
        'resolved_at',
    ];

    protected $casts = [
        'type' => DisputeType::class,
        'status' => DisputesStatus::class,
        'resolved_at' => 'datetime',
        'deleted_at' => 'datetime',
    ];

    protected $dates = [
        'deleted_at',
    ];

    use \Illuminate\Database\Eloquent\SoftDeletes;

    public function order()
    {
        return $this->belongsTo(Order::class);
    }

    public function customer()
    {
        return $this->belongsTo(User::class, 'customer_id');
    }

    public function seller()
    {
        return $this->belongsTo(User::class, 'seller_id');
    }

    public function assignedAdmin()
    {
        return $this->belongsTo(User::class, 'assigned_admin_id');
    }
}
